### PR TITLE
WIP: V5

### PR DESCRIPTION
**Pull request by twiss**

### CODE REVIEW

File: src/key/factory.js [hunk 1]
```diff
@@ -266,61 +263,67 @@ async function wrapKeyObject(secretKeyPacket, secretSubkeyPackets, options) {
...
-      }
-    }
-  } catch (e) {
-    err.push(e);
-  }
-  if (err.length) {
-    result.err = err;
+  const packetlist = new packet.List();
+  await packetlist.read(data);
+  return new Key(packetlist);
```
Comment (chesnokovilya) [inline src/key/factory.js:275]: It would throw if the key is bad or armored. Open questions if we should catch it?

Comment (twiss) [reply src/key/factory.js:275]: I think it's better to throw than to return an error array. Most people don't check it, and are confused when the `keys` array is empty.

**Comment by tomholub** [comment]
Just wanted to say, PRs that remove code are my favorite :smiley: 

image

**Comment by twiss** [comment]
Yes :smile: To be fair, a lot of that is removing the `dist` folder (people can get it from npm or unpkg.com instead), but still despite that we've removed quite some code as well :blush:

**Pull request merged**